docs/src/tutorial/minimal_working_compute.jl

```diff
@@ -0,0 +1,266 @@
+# `Float32`s, so that we can access it easily.
+data = unsafe_wrap(Vector{Float32}, convert(Ptr{Float32}, unwrap(memptr)), data_items, own = false)
+
+# For now, let's just zero out all the data, and *flush* the changes to make
+# sure the device can see the updated data. This is the simplest way to move
+# array data to the device.
+data .= 0
+flush_mapped_memory_ranges(device, [MappedMemoryRange(mem, 0, mem_size)])
+
+# Eventually, you may need to allocate memory types that are not visible from
+# host, because these provide better capacity and speed on the discrete GPUs.
+# At that point, you may need to use the transfer queue and memory transfer
+# commands to get the data from host-visible to GPU-local memory, using e.g.
+# [`cmd_copy_buffer`](@ref).
+
+# ## Compiling the shader
+#
+# Now we need to make a shader program. We will use `glslangValidator` packaged
+# in a JLL to compile a GLSL program from a string into a spir-v bytecode,
+# which is later passed to the GPU drivers.
+
+shader_code = """
+#version 430
+
+layout (local_size_x_id = 0) in;
+layout (local_size_x = 1, local_size_y = 1, local_size_z = 1) in;
+
+layout(constant_id = 0) const uint blocksize = 1; // copy of local_size_x
+
+layout (push_constant) uniform Params
+{
+    float val;
+    uint n;
+} params;
+
+layout(std430, binding=0) buffer databuf
+{
+    float data[];
+};
+
+void
+main()
+{
+    uint i = gl_GlobalInvocationID.x;
+    if(i < params.n) data[i] = params.val * i;
+}
+"""
+
+# Push constants are small packs of variables that are used to quickly send
+# configuration data to the shader runs. Make sure that this structure
+# corresponds to what is declared in the shader.
+struct shader_push_consts
+    val::Float32
+    n::UInt32
+end
+
+# Specialization constants are similar to push constants, but less dynamic: You
+# can change them before compiling the shader for the pipeline, but not
+# dynamically. This may have performance benefits for "very static" values,
+# such as block sizes.
+struct shader_spec_consts
+    local_size_x::UInt32
+end
+
+# Run the `glslangValidator` program to compile the shader.
+using glslang_jll
+glslang = glslangValidator(bin -> open(`$bin -V --quiet --stdin -S comp -o /dev/stdout `, read = true, write = true))
+write(glslang, shader_code) # send the code to stdin
+close(glslang.in) # send EOF
+shader_bcode = collect(reinterpret(UInt32, read(glslang))) #read the compiled shader spir-v code
+close(glslang)
+@assert glslang.exitcode == 0
+
+# Make a shader module with the code
+shader = unwrap(create_shader_module(device, sizeof(UInt32) * length(shader_bcode), shader_bcode))
+
+# ## Assembling the pipeline
+#
+# Descriptor set layout describes how many buffers of what kind are going to be
+# used by the shader
+dsl = unwrap(
+    create_descriptor_set_layout(
+        device,
+        [DescriptorSetLayoutBinding(0, DESCRIPTOR_TYPE_STORAGE_BUFFER, 1, SHADER_STAGE_COMPUTE_BIT, Sampler[])],
```

Comment:
```suggestion
        [DescriptorSetLayoutBinding(0, DESCRIPTOR_TYPE_STORAGE_BUFFER, SHADER_STAGE_COMPUTE_BIT; descriptor_count = 1)],
```

Seems more natural to me, but feel free to dismiss the suggestion if you prefer the original.

Reply:
cool, didn't know about that possibility.